### PR TITLE
JS API Error handling

### LINKED ISSUE #1065: Table Errors are Ignored
**Description**

If I have a table that produces an error, then the UI does not show it to me.

**Steps to reproduce**

1. Enter a command that will have a failed update into your console.  For example:
```
from deephaven.TableTools import timeTable
tt=timeTable("00:00:01").update("X=ii > 5 ? null : `z`", "Y=X.substring(0)")

```

Wait 6 seconds, until there are enough rows to trigger the NPE.  The table stops ticking, but there is no error indication. 

**Expected results**

On DHE the corresponding code is:
```
tt=db.timeTable("00:00:01").update("X=ii > 5 ? null : `z`", "Y=X.substring(0)")
```
table will have "Unable to open table. Error: Error displaying table." painted over it.

The actual proper behavior would be for the table display to tell me that there was a failure in updating and the original exception.

### PR DESCRIPTION
**Pull request by niloc132**
Added basic error handling, but not proper reconnects or reauth. Three error types are handled for the main streams or polling calls:
 * auth failure which suggests either that the server was restarted or the session timed out somehow - in either case, a new auth is required to resume working
 * Code.INTERNAL, which our grpc-web client library will pass in cases where the response doesn't make sense (broken payloads)
 * Code.UNKNOWN, which our grpc-web client library will pass in cases where the transport doesn't seem to work (no response before connection died)

A new event was added, temporarily, to inform the UI that we are giving up if one of these happens, and the page needs to be reloaded. This will be removed when we have proper reconnect, and inform the UI again that the reconnect succeeded or failed.

Also modified a few errors being sent from the server to hopefully better reflect the intent of the gRPC status codes, and avoid using "INTERNAL" when "something bad" happens.

The UI will need a few small changes to handle this, and to handle a few cases where API calls can fail but not be registered in the UI.

Fixes #1065, but #730 needs to formally address this.

### CODE REVIEW

**Comment by niloc132** [comment]
Extremely rough patch for @mofojed to finish up nicely in the UI https://github.com/niloc132/web-client-ui/commit/e798052121b5004ca2204d1a7aa7c50f86315b45

File: web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java [hunk 1]
```diff
@@ -65,6 +67,9 @@ static AuthTokenPromiseSupplier oneShot(ConnectToken initialToken) {
     @JsProperty(namespace = "dh.QueryInfo")
     public static final String EVENT_CONNECT = "connect";
 
+    @JsProperty(namespace = "dh.IdeConnection")
+    public static final String HACK_CONNECTION_FAILURE = "hack-connection-failure";
```
Comment (mofojed) [inline web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java:71]: Why is this named `hack-connection-failure`? Why `hack-`?

**Review by nbauernfeind — APPROVED** [review]

File: web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java (hunk 1 above)
Comment (nbauernfeind) [reply web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java:71]: I believe 'hack' here is to make it blindly obvious this is a temporary work around to a bigger problem that Colin wants carte blanch to fix correctly.

Comment (niloc132) [reply web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java:71]: This is temporary, until we have reconnect (#730) which will have this being removed. I wanted to draw a distinction between normal events or internal-only ones, but I'm open to some other way to do that.

I tried to summarize in the description above (and will make it part of the actual commit message), but the old reconnect events are probably what we will use for these cases once the api is smart enough to try reconnecting streams.

Comment (mofojed) [reply web/client-api/src/main/java/io/deephaven/web/client/api/QueryConnectable.java:71]: Okay cool, I've got a change to listen for these new errors pushed on my Application Mode PR: https://github.com/deephaven/web-client-ui/pull/155

**Review by mofojed — APPROVED** [review]

**Pull request merged**